kia-cli/src/main/kotlin/com/faforever/ice/KiaApplication.kt

```diff
@@ -77,7 +78,7 @@ class KiaApplication : Callable<Int> {
     private lateinit var iceAdapter: IceAdapter
     private lateinit var rpcService: RpcService
 
-    private fun onConnectionStateChanged(newState: String) = rpcService.onConnectionStateChanged(newState)
+    private fun onConnectionStateChanged(newState: GpgnetProxy.ConnectionState) = rpcService.onConnectionStateChanged(newState)
```

Comment:
`GpgnetProxy` ultimately calls this when the state changes. My change here makes it more generic (it takes an enum rather than the rpc service specific string) and it is always called when the state changes rather than just the cases handled by the rpc service. This allows the handler to also be used to send the `TelemetryClient.updateGpgnetState` message.